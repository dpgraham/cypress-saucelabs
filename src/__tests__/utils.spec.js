const { createProjectZip, startTunnel } = require("../utils");

jest.mock('fs');
jest.mock('adm-zip');
jest.mock('saucelabs');
const fs = require('fs');
const AdmZip = require('adm-zip');
const SauceLabs = require('saucelabs');

const mockLog = {
  info: () => {},
  error: () => {},
};

describe('utils', function () {
  describe('.createProjectZip', function () {
    it('should send non-ignored files to a zip entry', function () {
      fs.existsSync.mockImplementation(jest.fn(() => false));
      fs.readdirSync.mockImplementation(jest.fn((dir) => {
        if (dir === '/root/b/') {
          return ['c'];
        } else if (dir === '/root') {
          return ['a', 'b/', '.sauceignore', '.git']
        }
      }));
      fs.lstatSync.mockImplementation(jest.fn((dir) => ({isDirectory: () => dir.endsWith('/')})));
      fs.readFileSync.mockImplementation(() => 'hello world');
      AdmZip.prototype = {
        addLocalFile: jest.fn(),
        writeZip: jest.fn(),
      };
      createProjectZip({zipFileOut: '/path/to/zip', workingDir: '/root', log: mockLog});
      expect([AdmZip.prototype.addLocalFile.mock.calls, AdmZip.prototype.writeZip]).toMatchSnapshot();
    });
  });
  describe('.startTunnel', function () {
    it.only('should start a SauceConnect tunnel via SauceLabs NPM package', async function () {
      SauceLabs.default.prototype.startSauceConnect = jest.fn(() => 'some-tunnel-id');
      await startTunnel('fakeuser', 'fakeaccesskey', mockLog);
      const calls = SauceLabs.default.prototype.startSauceConnect.mock.calls;
      calls[0].logger();
      expect(calls).toMatchSnapshot();
    });
  });
});