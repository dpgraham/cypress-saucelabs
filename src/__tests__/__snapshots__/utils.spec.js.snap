// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils .createProjectZip should do something 1`] = `
Array [
  Array [
    Array [
      "/root/a",
      ".",
      "a",
    ],
    Array [
      "/root/b/c",
      "b",
      "c",
    ],
  ],
  [MockFunction] {
    "calls": Array [
      Array [
        "/path/to/zip",
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
]
`;

exports[`utils .createProjectZip should send non-ignored files to a zip entry 1`] = `
Array [
  Array [
    Array [
      "/root/a",
      ".",
      "a",
    ],
    Array [
      "/root/b/c",
      "b",
      "c",
    ],
  ],
  [MockFunction] {
    "calls": Array [
      Array [
        "/path/to/zip",
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
]
`;
